actions/lib/providers.py

```diff
@@ -42,3 +42,34 @@ def credentials_test(self, client, kwargs_dict):
                             'valid': valid})
 
         return (all_valid, results)
+
+    def refresh(self, client, kwargs_dict):
+        """Refresh the specified provider in ManageIQ. If no ID is given then
+         all providers will be refreshed.
+        :param client: ManageIQClient class object from manageiq_client.api
+        :param kwargs_dict: Inputs from the Stackstorm action
+        :returns: a list of dictionary results returned from ManageIQ
+        :rtype: list
+        """
+        providers = self._get_objects(client=client,
+                                      collection_name="providers",
+                                      query_dict={'expand': 'resources'})
+
+        resources = []
+        # If a provider ID was specified, only refresh that one
+        if kwargs_dict['provider_id']:
+            for prov in providers:
+                if str(prov['id']) == kwargs_dict['provider_id']:
+                    resources.append(prov)
+                    break
+        else:
+            resources = providers
+
+        result = client.collections.providers.action.refresh(*resources)
+
+        # return the data objects from each of the action results
+        results = []
```

Comment:
could probably use this: _data_from_entity_list()
https://github.com/StackStorm-Exchange/stackstorm-manageiq/blob/master/actions/lib/base_action.py#L44-L48